dmenu.el

```diff
@@ -81,18 +81,22 @@ Must be set before initializing Dmenu."
   (let* ((completing-read-fn (if ido-mode
                                  #'ido-completing-read
                                #'completing-read))
-         (execute-file (funcall completing-read-fn dmenu-prompt-string
-                                        (append dmenu--history-list
-                                                (cl-remove-if (lambda (x)
-                                                                (member x dmenu--history-list))
-                                                              dmenu--cache-executable-files))
-                                        nil
-                                        'confirm
-                                        nil
-                                        'dmenu--history-list))
+         (execute-file (funcall completing-read-fn
+                                dmenu-prompt-string
+                                (append dmenu--history-list
+                                        dmenu--cache-executable-files)
```

Comment:
I don't understand why changes here?

Reply:
(My comment on the other code chunk applies here.)

Since the new code does NOT cache commands without arguments in `dmenu--history-list` anymore, we do not therefore need to filter them out of `dmenu--history-list` (with `cl-remove-if`) as before:
```
(append dmenu--history-list
        (cl-remove-if (lambda (x)
                        (member x dmenu--history-list))
                      dmenu--cache-executable-files))
```
That code now simply becomes:
```
(append dmenu--history-list
        dmenu--cache-executable-files)
```